serialization/bigint/unmarshal_utils.go

```diff
@@ -398,7 +407,10 @@ func DecUintR(p []byte, v **uint) error {
 			*v = new(uint)
 		}
 	case 8:
-		val := uint(p[0])<<56 | uint(p[1])<<48 | uint(p[2])<<40 | uint(p[3])<<32 | uint(p[4])<<24 | uint(p[5])<<16 | uint(p[6])<<8 | uint(p[7])
+		if p[0] > math.MaxInt8 {
```

Comment:
Why do you consider only first byte ? It looks like mistake

Reply:
The `p[0] > math.MaxInt8` condition is are like a question `is a data value in the negative range?`